RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -899,420 +955,400 @@ class DeepTauId : public deep_tau::DeepTauBase {
+            get(dnn::pfCand_chHad_vertex_dz_tauFL) = getValueNorm(pfCands.at(index_chH).vertex().z() - pv.position().z()
+                - tau.flightLength().z(), -0.0138f, 8.622f);
+
+            const bool hasTrackDetails = pfCands.at(index_chH).hasTrackDetails();
+            if(hasTrackDetails){
+                get(dnn::pfCand_chHad_hasTrackDetails) = hasTrackDetails;
+                get(dnn::pfCand_chHad_dxy) = getValueNorm(pfCands.at(index_chH).dxy(), -0.012f, 2.386f);
+                get(dnn::pfCand_chHad_dxy_sig) = getValueNorm(std::abs(pfCands.at(index_chH).dxy()) /
+                    pfCands.at(index_chH).dxyError(), 6.417f, 36.28f);
+                get(dnn::pfCand_chHad_dz) =  getValueNorm(pfCands.at(index_chH).dz(), -0.0246f, 7.618f);
+                get(dnn::pfCand_chHad_dz_sig) =  getValueNorm(std::abs(pfCands.at(index_chH).dz()) /
+                    pfCands.at(index_chH).dzError(), 301.3f, 491.1f);
+                get(dnn::pfCand_chHad_track_chi2_ndof) = pfCands.at(index_chH).pseudoTrack().ndof() > 0 ?
+                    getValueNorm(pfCands.at(index_chH).pseudoTrack().chi2() /
+                    pfCands.at(index_chH).pseudoTrack().ndof(), 0.7876f, 3.694f) : 0;
+                get(dnn::pfCand_chHad_track_ndof) = pfCands.at(index_chH).pseudoTrack().ndof() > 0 ?
+                    getValueNorm(pfCands.at(index_chH).pseudoTrack().ndof(), 13.92f, 6.581f) : 0;
             }
+            get(dnn::pfCand_chHad_hcalFraction) = getValue(pfCands.at(index_chH).hcalFraction());
+            get(dnn::pfCand_chHad_rawCaloFraction) = getValueLinear(pfCands.at(index_chH).rawCaloFraction(), 0.f, 2.6f, true);
+        }
+        if(valid_nH){
+            size_t index_nH = cell_map.at(CellObjectType::PfCand_neutralHadron);
+
+            get(dnn::pfCand_nHad_valid) = valid_nH;
+            get(dnn::pfCand_nHad_rel_pt) = getValueNorm(pfCands.at(index_nH).polarP4().pt() / tau.polarP4().pt(),
+                is_inner ? 0.3163f : 0.0502f, is_inner ? 0.2769f : 0.4266f);
+            get(dnn::pfCand_nHad_deta) = getValueLinear(pfCands.at(index_nH).polarP4().eta() - tau.polarP4().eta(),
+                is_inner ? -0.1f : -0.5f, is_inner ? 0.1f : 0.5f, false);
+            get(dnn::pfCand_nHad_dphi) = getValueLinear(dPhi(tau.polarP4(),pfCands.at(index_nH).polarP4()),
+                is_inner ? -0.1f : -0.5f, is_inner ? 0.1f : 0.5f, false);
+            get(dnn::pfCand_nHad_puppiWeight) = getValue(pfCands.at(index_nH).puppiWeight());
+            get(dnn::pfCand_nHad_puppiWeightNoLep) = getValue(pfCands.at(index_nH).puppiWeightNoLep());
+            get(dnn::pfCand_nHad_hcalFraction) = getValue(pfCands.at(index_nH).hcalFraction());
         }
-        checkInputs(inputs, is_inner ? "hadron_inner_block" : "hadron_outer_block", dnn::NumberOfInputs, grid.nCellsEta,
-            grid.nCellsPhi);
+        // checkInputs(inputs, is_inner ? "hadron_inner_block" : "hadron_outer_block", dnn::NumberOfInputs, grid.nCellsEta,
```

Comment:
Is there any important reason to keep this code commented out? If, so please comment on this in code otherwise remove.